contracts/dydx-airdrop/src/validation.rs

```diff
@@ -0,0 +1,205 @@
+        false => Err(ContractError::AddressNotEligible {
+            address: eth_address,
+        }),
+    }
+}
+
+fn validate_eth_sig(
+    deps: &DepsMut,
+    info: MessageInfo,
+    eth_address: String,
+    eth_sig: String,
+    config: Config,
+) -> Result<(), ContractError> {
+    let valid_eth_sig = validate_ethereum_text(deps, info, &config, eth_sig, eth_address.clone())?;
+    match valid_eth_sig {
+        true => Ok(()),
+        false => Err(ContractError::AddressNotEligible {
+            address: eth_address,
+        }),
+    }
+}
+
+pub fn validate_collection_mint(deps: &DepsMut, info: MessageInfo) -> Result<(), ContractError> {
+    let collection_address = query_collection_address(deps)?;
+    let tokens_response: TokensResponse = deps.querier.query_wasm_smart(
+        collection_address,
+        &QueryMsg::Tokens {
+            owner: String::from(info.sender.clone()),
+            start_after: None,
+            limit: None,
+        },
+    )?;
+    return if tokens_response.tokens.len() == 0 {
+        Err(ContractError::CollectionNotMinted {})
+    } else {
+        Ok(())
+    };
+}
+
+pub fn validate_name_mint_and_association(
+    deps: &DepsMut,
+    info: MessageInfo,
+) -> Result<(), ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+    let name_collection_address = config.name_collection_address;
+    let tokens_response: TokensResponse = deps.querier.query_wasm_smart(
+        name_collection_address.clone(),
+        &QueryMsg::Tokens {
+            owner: String::from(info.sender.clone()),
+            start_after: None,
+            limit: None,
+        },
+    )?;
+    if tokens_response.tokens.len() == 0 {
+        return Err(ContractError::NameNotMinted {});
+    };
+    let _associated_name: String = deps.querier.query_wasm_smart(
+        name_collection_address.clone(),
+        &NameQueryMsg::Name {
+            address: String::from(info.sender.clone()),
+        },
+    )?;
+    Ok(())
+}
+
+pub fn validate_ethereum_text(
+    deps: &DepsMut,
+    info: MessageInfo,
+    config: &Config,
+    eth_sig: String,
+    eth_address: String,
+) -> StdResult<bool> {
+    let plaintext_msg = compute_plaintext_msg(config, info);
+    match hex::decode(eth_sig.clone()) {
+        Ok(eth_sig_hex) => {
+            verify_ethereum_text(deps.as_ref(), &plaintext_msg, &eth_sig_hex, &eth_address)
+        }
+        Err(_) => Err(StdError::InvalidHex {
+            msg: format!("Could not decode {eth_sig}"),
+        }),
+    }
+}
+
+pub fn check_previous_registration(deps: &DepsMut, eth_address: &str) -> Result<(), ContractError> {
+    let registered = IS_ADDRESS_REGISTERED
+        .load(deps.storage, eth_address)
+        .unwrap_or(false);
+    if registered {
+        return Err(ContractError::AlreadyRegistered {
+            address: eth_address.to_string(),
+        });
+    }
+    Ok(())
+}
+
+pub fn check_previous_claim(deps: &DepsMut, eth_address: &str) -> Result<(), ContractError> {
```

Comment:
Shouldn't a "previous claim" just be the address to mint count matches the max airdrop amount or do we need to track something different?

Reply:
Only the reward is being claimed currently. Not tracking a mint count. The user is expected to mint the token outside of the airdrop contract through the regular minting process.